packages/extension-utils/src/OAuthConfigForm/OAuthConfigForm.tsx

```diff
@@ -119,54 +96,28 @@ export const OAuthConfigForm = ({
   }
 
   const config = getConfig()
-  const [fields, setFields] = useState<IFieldValues>(defaultFieldValues)
-  const [saved, setSaved] = useState<ConfigValues>(config)
-
-  const updateFields = (
-    nameOrValues: string | Partial<IFieldValues>,
-    value = ''
-  ) => {
-    if (typeof nameOrValues === 'string') {
-      setFields((previousFields) => {
-        return { ...previousFields, ...{ [nameOrValues]: value } }
-      })
-    } else {
-      setFields((previousFields) => {
-        return { ...previousFields, ...nameOrValues }
-      })
-    }
-  }
-
-  useEffect(() => {
-    const data = getConfig()
-    const { base_url, looker_url } = data
-    setSaved(data)
-    updateFields({
-      [BASE_URL]: base_url,
-      [WEB_URL]: looker_url,
-      [FETCH_INTENT]:
-        base_url !== '' && looker_url !== '' ? POSITIVE : CRITICAL,
-    })
-  }, [])
-
+  const [apiServerUrl, setApiServerUrl] = useState('')
+  const [fetchedUrl, setFetchedUrl] = useState('')
+  const [webUrl, setWebUrl] = useState('')
+  const [messageBarIntent, setMessageBarIntent] =
+    useState<MessageBarIntent>('positive')
+  const [messageBarText, setMessageBarText] = useState('')
   const [validationMessages, setValidationMessages] =
     useState<ValidationMessages>({})
+  const [saved, setSaved] = useState<ConfigValues>(config)
 
-  const updateMessage = (intent: MessageBarIntent, message: string) => {
-    updateFields({ [FETCH_RESULT]: message, [FETCH_INTENT]: intent })
+  const updateMessageBar = (intent: MessageBarIntent, message: string) => {
+    setMessageBarIntent(intent)
+    setMessageBarText(message)
   }
 
   const isConfigured = () => {
-    return (
-      saved !== EmptyConfig &&
-      fields[BASE_URL] === saved.base_url &&
-      fields[WEB_URL] === saved.looker_url
-    )
+    return apiServerUrl === saved.base_url && webUrl === saved.looker_url
```

Comment:
comparing the EmptyConfig always results in false, as objects are passed / compared by reference in JS.

Reply:
I'm guessing empty urls can't be saved?